Modules/_elementtree.c

```diff
@@ -109,11 +106,17 @@ get_elementtree_state(PyObject *module)
     return (elementtreestate *)state;
 }
 
-/* Find the module instance imported in the currently running sub-interpreter
- * and get its state.
- */
-#define ET_STATE_GLOBAL \
-    ((elementtreestate *) PyModule_GetState(PyState_FindModule(&elementtreemodule)))
+static inline elementtreestate *
+get_elementtree_module_state_by_object(void *object)
```

Comment:
The qualifier ``object`` is not self-explanatory. Every Python object is an object. This includes module objects and type objects. How about:

```suggestion
get_elementtree_module_state_by_instance(PyObject *self)
```

Reply:
How about expend `get_element_module_state()` to receive `self` object or `type` object?

Reply:
@tiran I tried using `PyObject *` and that gives a lot of compiler warnings (see somem below):

>>> C:\Dev\OSS\cpython\cpython\Modules\_elementtree.c(3749,74): warning C4133: 'function': incompatible types - from 'XMLParserObject *' to 'PyObject *' [C:\Dev\OSS\cpython\cpython\PCbuild\_elementtree.vcxproj]
C:\Dev\OSS\cpython\cpython\Modules\clinic\_elementtree.c.h(26,52): warning C4133: 'function': incompatible types - from 'ElementObject *' to 'PyObject *' [C:\Dev\OSS\cpython\cpython\PCbuild\_elementtree.vcxproj]

Reply:
@shihai1991 do you mean two methods, one for each?

Reply:
yes. just my personal suggestion :)